### PR TITLE
Fixes #25594 - issue warning during unused_ip call

### PR DESCRIPTION
**Pull request by lzap**
This took me some time today to figure out, taxonomy issue.

### CODE REVIEW

File: app/services/ipam/dhcp.rb [hunk 1]
```diff
@@ -2,7 +2,10 @@ module IPAM
   class Dhcp < Base
     delegate :dhcp, :dhcp_proxy, :to => :subnet
     def suggest_ip
-      return unless subnet.dhcp?
+      unless subnet.dhcp?
+        logger.warn "Unable to suggest IP: subnet #{subnet.name} don't have a DHCP proxy associated, a proxy is not in subnets organization or IPAM is not set to DHCP"
```
Comment (ekohl) [inline app/services/ipam/dhcp.rb]: don't -> doesn't
the subnets organization
I'd also consider writing organization/location

**Review by ekohl — COMMENTED** [review]
I do wonder if this is really a warning. Warning implies a misconfiguration but this code can also trigger in legitimate situations. If a subnet has no DHCP proxy attached, the free ip will be called regardless and spam the logs with warnings. Perhaps it's more an info level?

File: app/services/ipam/dhcp.rb (hunk 1 above)
Comment (lzap) [reply app/services/ipam/dhcp.rb]: Done

**Comment by tbrisker** [comment]
ping @lzap

**Comment by lzap** [comment]
@ekohl rebased, the message is now info level.

File: app/services/ipam/dhcp.rb [hunk 2]
```diff
@@ -2,7 +2,10 @@ module IPAM
   class Dhcp < Base
     delegate :dhcp, :dhcp_proxy, :to => :subnet
     def suggest_ip
-      return unless subnet.dhcp?
+      unless subnet.dhcp?
+        logger.info "Unable to suggest IP: no DHCP proxy or IPAM set for subnet '#{subnet.name}' or taxonomy does not match host"
+        return
```
Comment (tbrisker) [inline app/services/ipam/dhcp.rb:9]: Should we also add an error here (like we do below) so the user sees this in the UI and not just the logs?

Comment (lzap) [reply app/services/ipam/dhcp.rb:9]: Very good idea actually! I am also improving code here and below so we DRY.

File: app/services/ipam/dhcp.rb [hunk 3]
```diff
@@ -2,15 +2,21 @@ module IPAM
   class Dhcp < Base
     delegate :dhcp, :dhcp_proxy, :to => :subnet
     def suggest_ip
-      return unless subnet.dhcp?
+      unless subnet.dhcp?
+        message = N_('Failed to fetch a free IP from proxy %{proxy}: %{message}') % {:message => e, :proxy => "#{dhcp.name} (#{dhcp.url})"}
```
Comment (tbrisker) [inline app/services/ipam/dhcp.rb]: Looks like copy-paste got you? wasn't this supposed to be something like
```suggestion
        message = N_("Unable to suggest IP: subnet %s don't have a DHCP proxy associated, a proxy is not in subnets organization or IPAM is not set to DHCP") % subnet.name
```

File: app/services/ipam/dhcp.rb [hunk 4]
```diff
@@ -2,15 +2,21 @@ module IPAM
...
+        logger.warn message
```
Comment (tbrisker) [inline app/services/ipam/dhcp.rb]:
```suggestion
        logger.info message
```

**Comment by lzap** [comment]
Oh yeah, that was it. Rebased, thanks.

**Review by tbrisker — COMMENTED** [review]
LGTM! when testing i noticed that we have a similar issue with subnet that has ipam set to none - care to fix it there on the way?

**Comment by lzap** [comment]
What exactly do you see? Because IPAM set to None is actually a useful setting. It means that no IPAM is performed therefore user can enter IP address manually. Do you see an error?

**Comment by tbrisker** [comment]
@lzap I mean i don't see any message in the ui indicating the subnet has no ipam configured, which confused me as i was expecting the new message to appear (i forgot to set the subnet to dhcp ipam). so i was thinking maybe we should display a similar message in that case as well?

**Review by tbrisker — DISMISSED** [review]
Thanks @lzap ! I'm fine with keeping the None one separate from this pr.

**Comment by lzap** [comment]
Rebased.

**Comment by tbrisker** [comment]
test failure unrelated.

**Pull request merged**